tracer/src/Datadog.Trace/Util/SamplingHelpers.cs

```diff
@@ -16,5 +17,22 @@ internal static class SamplingHelpers
 
         internal static bool IsKeptBySamplingPriority(ArraySegment<Span> trace) =>
             trace.Array![trace.Offset].Context.TraceContext?.SamplingPriority > 0;
+
+        internal static int? GetDecisionMaker(TraceContext traceContext)
+        {
+            var decisionMaker = traceContext.Tags.GetTag(Tags.Propagated.DecisionMaker);
+
+            if (decisionMaker != null)
+            {
+                var hyphen = decisionMaker.IndexOf('-');
+
+                if (hyphen != -1 && int.TryParse(decisionMaker.Substring(hyphen + 1), out var value))
```

Comment:
Shouldn't we store the mechanism instead in TraceContext? I assume it's one extra field so more memory on each Trace for a feature we don't use much, but it maybe more straightforward and future proof? 🤷

Reply:
This is probably what we should do, now that we need to parse it it. Up until now we only needed to either propagate the tag we got form upstream, or add our own if the trace is kept.

This will also solves the issue of the tag _not_ being present if the trace is not sampled (sampling priority < 1).